x/auth/keeper/keeper.go

```diff
@@ -145,6 +148,11 @@ func NewAccountKeeper(
 	return ak
 }
 
+// GetAccountNumber returns the x/auth module's current account number.
+func (ak AccountKeeper) GetAccountNumber(ctx context.Context) (uint64, error) {
```

Comment:
This method is called only once for migration purpose. It would be good make this clear in the naming so that this method is not used otherwise. You can even go one step further and remove the entry in the DB for cleanup. 
```suggestion
// RemoveLegacyAccountNumber is used for migration purpose only. It deletes the sequence in the DB
// and returns the last value used on success.
// Deprecated
func (ak AccountKeeper) RemoveLegacyAccountNumber(ctx context.Context) (uint64, error) {
// peek
// delete
```

Reply:
updated as suggestion